scripts/bert/finetune_classifier.py

```diff
@@ -346,13 +362,64 @@ def train(metric):
                     ','.join([i + ':{:.4f}' for i in metric_nm])
                 logging.info(eval_str.format(epoch_id + 1, batch_id + 1, len(train_data), \
                     step_loss / args.log_interval, \
-                    trainer.learning_rate, *metric_val))
+                    trainer.learning_rate, *metric_val,))
                 step_loss = 0
+
         mx.nd.waitall()
         evaluate(metric)
         toc = time.time()
         logging.info('Time cost={:.1f}s'.format(toc - tic))
-        tic = toc
+
+        # Save parameters
+        validation_acc = metric.get()[1][0]
+        if validation_acc > best_acc:
+            save_path = os.path.join(args.save_dir, 'valid_best.params')
+            model.save_parameters(save_path)
+            best_acc = validation_acc
+            logging.info('the best acc is updata, and valid_best params saved as:{}'.format(save_path))
+
+
+def inference(metric):
+    """inference function."""
+
+    logging.info('|----- Now we are doing BERT inference at {} !'.format(ctx))
+    model = BERTClassifier(
+        bert, dropout=0.1, num_classes=len(task.get_labels()))
+    para_name = 'valid_best.params'
+    model.load_parameters(os.path.join(args.save_dir, para_name), ctx=ctx)
+
+    for epoch_id in range(1):
+        metric.reset()
+        step_loss = 0
+        tic = time.time()
+
+        for batch_id, seqs in enumerate(train_data):
+
+            input_ids, valid_length, type_ids, label = seqs
+            out = model(
+                input_ids.as_in_context(ctx), type_ids.as_in_context(ctx),
+                valid_length.astype('float32').as_in_context(ctx))
+            ls = loss_function(out, label.as_in_context(ctx)).mean()
+
+            step_loss += ls.asscalar()
+            metric.update([label], [out])
+
+            if (batch_id + 1) % (args.log_interval) == 0:
+                metric_nm, metric_val = metric.get()
+                if not isinstance(metric_nm, list):
+                    metric_nm = [metric_nm]
+                    metric_val = [metric_val]
+                eval_str = '[Epoch {} Batch {}/{}] loss={:.4f}, metrics=' + \
+                    ','.join([i + ':{:.4f}' for i in metric_nm])
+                logging.info(eval_str.format(epoch_id + 1, batch_id + 1, len(train_data), \
+                    step_loss / args.log_interval, \
+                    *metric_val,))
+                step_loss = 0
+
+        mx.nd.waitall()
+
+    toc = time.time()
+    logging.info('Time cost={:.1f}s'.format(toc - tic))
```

Comment:
move this to for loop? we need measure the time of each epoch like what did in train function.